File: clients/csharp/Model/ColoredVertex.cs
namespace AiCup2019.Model
{
    public struct ColoredVertex
    {
        public Model.Vec2Float Position { get; set; }
        public Model.ColorFloat Color { get; set; }
        public ColoredVertex(Model.Vec2Float position, Model.ColorFloat color)
        {
            this.Position = position;
            this.Color = color;
        }
        public static ColoredVertex ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new ColoredVertex();
            result.Position = Model.Vec2Float.ReadFrom(reader);
            result.Color = Model.ColorFloat.ReadFrom(reader);
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            Position.WriteTo(writer);
            Color.WriteTo(writer);
        }
    }
}

File: clients/fsharp/Model/ColoredVertex.fs
#nowarn "0058"
namespace AiCup2019.Model
type ColoredVertex = {
    Position: Vec2Single;
    Color: ColorSingle;
    } with
    member this.writeTo(writer: System.IO.BinaryWriter) =
        this.Position.writeTo writer
        this.Color.writeTo writer
    static member readFrom(reader: System.IO.BinaryReader) = {
        Position = Vec2Single.readFrom reader
        Color = ColorSingle.readFrom reader
    }
